cmd/dev/ci/deps/url_test.go

```diff
@@ -63,11 +62,11 @@ func TestCustomArchURL(t *testing.T) {
 	_ = comp1.getComponentFromConfig("test/customArchURL.yaml")
 
 	url, err := comp1.getRenderedURL("darwin", "amd64")
-	assert.Nil(t, err, "Expected no Error!")
+	require.NoError(t, err)
```

Comment:
@alnr @CaptainStandby always use `require.NoError` when testing whether a function returns a nil error type. `require` prevents further test execution which is important as the first return value is typically nil if error is not nil. That can end up in a panic if we follow the execution branch.

`assert` will fire a test error, but will keep the execution of the test going. This is useful when you test a bunch of assertions after one another and you don't expect that one of them failing may end up in a panic

Reply:
makes sense

Reply:
I saw that the original code is way older and this was more or less copy&paste, but since I saw it in some other PR too I thought I'd point it out :)

Reply:
I've added this to the go style guide in notion